core-web/libs/portlets/dot-ema/src/lib/feature/dot-ema.component.ts

```diff
@@ -1,11 +1,110 @@
-import { CommonModule } from '@angular/common';
-import { Component } from '@angular/core';
+import { CommonModule, DOCUMENT } from '@angular/common';
+import { AfterViewInit, Component, ElementRef, Inject, ViewChild } from '@angular/core';
+
+import { Dialog, DialogModule } from 'primeng/dialog';
+
+import { DotCMSContentlet } from '@dotcms/dotcms-models';
+
+import { CUSTOM_EVENTS, MESSAGE_ACTIONS } from '../shared/models';
 
 @Component({
     selector: 'dot-ema',
     standalone: true,
-    imports: [CommonModule],
+    imports: [CommonModule, DialogModule],
     templateUrl: './dot-ema.component.html',
     styleUrls: ['./dot-ema.component.scss']
 })
-export class DotEmaComponent {}
+export class DotEmaComponent implements AfterViewInit {
+    @ViewChild('dialog') dialog!: Dialog;
+    @ViewChild('dialogIframe') dialogIframe!: ElementRef<HTMLIFrameElement>;
+    @ViewChild('iframe') iframe!: ElementRef<HTMLIFrameElement>;
+
+    visible = false;
+    header = '';
+
+    readonly host = 'http://localhost:3000';
+
+    constructor(@Inject(DOCUMENT) private document: Document) {}
+
+    ngAfterViewInit(): void {
+        this.document.defaultView?.addEventListener('message', (event: MessageEvent) => {
+            // This should be the host the user uses for nextjs, because this can trigger react dev tools messages
+            if (event.origin !== this.host) {
+                return;
+            }
+
+            this.handlePostMessage(event.data)();
+        });
```

Comment:
use `window.addEventListener` and you don't need this hook.

Reply:
It's safer to use the injection of the document, `window` object could be problematic or hard to debug is something happens

Reply:
What about this:

<img width="704" alt="image" src="https://github.com/dotCMS/core/assets/751424/4a1d7b57-4b73-49fb-8a22-11a6326e8cfa">

My concern is that from nextjs we are doing `window.parent.postMessage`, `parent` is the `window` not the document and in the past I've seen issues when you emit the event to the window, but you listen to it on the document.